src/util/mutationAlgorithms.ts

```diff
@@ -763,3 +764,189 @@ export function setTextContentByReplacing(contextObject: Node, newValue: string)
 	// 3. Replace all with node within the context object.
 	replaceAllWithNode(node, contextObject);
 }
+
+/**
+ * To convert nodes into a node, given nodes and document, run these steps:
+ *
+ * @param nodes    Nodes and/or strings to convert
+ * @param document Document used to create new nodes
+ *
+ * @returns A single node representing nodes
+ */
+function convertNodesIntoNode(nodes: (Node | string)[], document: Document): Node {
+	// 1. Let node be null.
+	// (created as needed in branches below)
+
+	// 2. Replace each string in nodes with a new Text node whose data is the string and node
+	// document is document.
+	const actualNodes: Node[] = (nodes = nodes.map(nodeOrString => {
```

Comment:
additional assignment to nodes?

Reply:
Oops, must have left that in by accident when I decided to introduce a new variable for better typing. Nice catch!